httplib_test.go

```diff
@@ -45,6 +45,7 @@ func TestReplyJSON(t *testing.T) {
 
 	testCase(t, errors.New("test error"))
 	testCase(t, &TraceErr{Err: errors.New("test error")})
+	testCase(t, &TraceErr{Err: errors.New("test error"), Traces: Traces{{Path: "A", Func: "B", Line: 1}}})
```

Comment:
Not strictly required, but we really should use `require.JSONEq` instead of `require.Equal` in the test.

Additionally, I'd probably rewrite this to use as a table test. With this implementation, if any of the 3 test cases fails, you're going to see the same error message pointing at line 43 and not knowing which test case it was.

Suggestion:
```
	for _, test := range []struct{
		desc string
		err error
	}{
		{"plain error", errors.New("test error")},
		{"trace error", &TraceError{Err: errors.New("test error")}},
		{"trace error with traces", &TraceError{Err: errors.New("test error"), Traces: Traces{{Path: "A", Func: "B", Line: 1}}},
	} {
		t.Run(test.desc, func (t *testing.T) {
			recorder := httptest.NewRecorder()
			replyJSON(recorder, errCode, test.err)
			require.JSONEq(t, expectedErrorResponse, recorder.Body.String())
		})
	}
```